statesync/stateprovider.go

```diff
@@ -194,6 +194,10 @@ func (s *lightClientStateProvider) State(ctx context.Context, height uint64) (sm
 	state.ConsensusParams = result.ConsensusParams
 	state.LastHeightConsensusParamsChanged = currentLightBlock.Height
 
+	// TODO may need to assign timeouts?
+	// state.TimeoutPropose
+	// state.TimeoutCommit
```

Comment:
[blocking] this needs to be addressed before merging

Reply:
To provide some context on why we haven't assigned timeouts yet (i.e., the TODO Rootul mentioned): I opened the PR early without fully testing state sync, as I was requested to do so. The TODO I added was mainly a personal reminder for future iterations of the PR, as well as to gather reviewers' opinions about it in this PR.

The reason that the lack of timeout assignments in the state sync may be acceptable is that: In the worst-case scenario, when a node completes state sync and reaches the chain tip, if the state it obtains has no timeouts (i.e., zero values), the node will fall back to using timeouts from its configuration (not application-dependent) during that current height. However, at the `EndBlock` of that height, it will receive the correct timeouts from the ABCI response and apply them for the next height.

That said, we might be able to avoid this situation based on Callum’s comment [here](https://github.com/celestiaorg/celestia-core/pull/1494#pullrequestreview-2362507406). I’m reviewing the code to fully understand the state sync interaction and will make the necessary adjustments to ensure the timeouts are set correctly, thus addressing the case mentioned above.

Reply:
Yes the timeouts in state should be set later when `Info` gets called

Reply:
resolved in https://github.com/celestiaorg/celestia-core/pull/1494/commits/12639d4aae7394824ea66df8190a16295e75018a